daphne/src/testing.rs

```diff
@@ -221,18 +221,67 @@ impl DapAggregator<BearerToken> for MockAggregator {
 
     async fn get_agg_share(
         &self,
-        _task_id: &Id,
-        _batch_interval: &Interval,
+        task_id: &Id,
+        batch_interval: &Interval,
     ) -> Result<DapAggregateShare, DapError> {
-        unreachable!("not implemented");
+        let task_config = self
+            .get_task_config_for(task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        let mut agg_share = DapAggregateShare::default();
```

Comment:
nit: For style I would move this declaration just before `agg_share` is first used. (This is the `for` loop below.)

Reply:
Great catch! I've moved this just before the for loop.